app/src/main/java/io/github/incplusplus/potwhole/MyAccountFragment.java

```diff
@@ -56,6 +73,79 @@ public void onCreate(Bundle savedInstanceState) {
     public View onCreateView(
             LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_my_account, container, false);
+
+        if (mAuth.getCurrentUser() == null) {
+            return inflater.inflate(R.layout.fragment_my_account_not_signed_in, container, false);
+        } else {
+            return inflater.inflate(R.layout.fragment_my_account_signed_in, container, false);
+        }
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        if (mAuth.getCurrentUser() == null) {
+
+            Button logInButton = getView().findViewById(R.id.log_in_button);
+
+            createAccount = (TextView) getView().findViewById(R.id.createAccount);
+
+            editTextEmail = getView().findViewById(R.id.editTextEmail);
+            editTextPassword = getView().findViewById(R.id.editTextPassword);
+
+            logInButton.setOnClickListener(v -> userLogin());
+
+            createAccount.setOnClickListener(
+                    v -> {
+                        Intent intent = new Intent(getActivity(), RegisterAccountPage.class);
+                        startActivity(intent);
+                    });
+
+        } else {
+            Button signOutButton = getView().findViewById(R.id.sign_out_button);
+
+            signOutButton.setOnClickListener(
+                    v -> {
+                        mAuth.signOut();
+
+                        Intent intent = new Intent(getActivity(), MainActivity.class);
+                        startActivity(intent);
+                    });
+        }
+    }
+
+    private void userLogin() {
+        String email = editTextEmail.getText().toString().trim();
+        String password = editTextPassword.getText().toString().trim();
+
+        if (email.isEmpty()) {
+            editTextEmail.setError("Please Enter an Email Address");
+            editTextEmail.requestFocus();
+            return;
+        }
+
+        if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
+            editTextEmail.setError("Please Enter a Valid Email Address");
+            editTextEmail.requestFocus();
+            return;
+        }
+
+        if (password.isEmpty()) {
+            editTextEmail.setError("Please Enter a Password");
+            editTextEmail.requestFocus();
+            return;
+        }
+
+        mAuth.signInWithEmailAndPassword(email, password)
+                .addOnCompleteListener(
+                        task -> {
+                            if (task.isSuccessful()) {
+                                // redirect
+                                System.out.println("User is Signed in");
```

Comment:
If you want to keep this around for debugging purposes, this should be switched out for a log statement.